File: hw/vargas/u3/hw26GUIs/java/Painting/Painting/src/main/java/ec/edu/espe/painting/utils/MongoConnection.java
package ec.edu.espe.painting.utils;
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoDatabase;    
/**
 *
 * @author César Vargas, Paradigm, @ESPE
 */
public class MongoConnection {

    private static MongoConnection instance;
    private MongoClient mongoClient;
    private MongoDatabase database;

    private static final String CONNECTION_STRING = "mongodb+srv://Cesar:Cesar2006@cluster0.tgbv2qc.mongodb.net/";
    private static final String DATABASE_NAME = "GalleryDB";

    private MongoConnection() {
        try {
            mongoClient = MongoClients.create(CONNECTION_STRING);
            database = mongoClient.getDatabase(DATABASE_NAME);
            System.out.println("Conexión a MongoDB exitosa.");
        } catch (Exception e) {
            System.err.println("Error conectando a MongoDB: " + e.getMessage());
        }
    }
    public static MongoConnection getInstance() {
        if (instance == null) {
            instance = new MongoConnection();
        }
        return instance;
    }

    public MongoDatabase getDatabase() {
        return database;
    }

    public void close() {
        if (mongoClient != null) {
            mongoClient.close();
            System.out.println("Conexión a MongoDB cerrada.");
        }
    }
}

File: hw/cevallos/u3/hw35StrategyPersistence/JavaScript/calendar-system/utils/MongoConnection.js
const { MongoClient } = require('mongodb');

class MongoConnection {
    static instance = null;
    static client = null;
    static database = null;

    static async connect() {
        if (!MongoConnection.client) {
            try {
                const uri = 'mongodb://localhost:27017';
                MongoConnection.client = new MongoClient(uri);
                
                await MongoConnection.client.connect();
                MongoConnection.database = MongoConnection.client.db('calendar_db');
                
                console.log('✅ Conectado a MongoDB exitosamente');
            } catch (error) {
                console.error('❌ Error conectando a MongoDB:', error.message);
                throw error;
            }
        }
        return MongoConnection.database;
    }

    static async getCollection(collectionName) {
        if (!MongoConnection.database) {
            await MongoConnection.connect();
        }
        return MongoConnection.database.collection(collectionName);
    }

    static async close() {
        if (MongoConnection.client) {
            await MongoConnection.client.close();
            MongoConnection.client = null;
            MongoConnection.database = null;
            console.log('🔌 Conexión a MongoDB cerrada');
        }
    }

    static getStatus() {
        return {
            connected: !!MongoConnection.client,
            database: MongoConnection.database ? MongoConnection.database.databaseName : null
        };
    }
}

module.exports = MongoConnection;